maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java

```diff
@@ -375,9 +454,7 @@ private ModelSource createStubModelSource( Artifact artifact )
 
         boolean noErrors =
             build( results, interimResults, projectIndex, pomFiles, new LinkedHashSet<>(), true, recursive,
-                   config );
-
-        populateReactorModelPool( modelPool, interimResults );
+                   config, poolBuilder );
```

Comment:
Why do you pass around and reuse the builder? Have the feeling your are misusing a build like a caching factory.

Reply:
The reactormodelpool was always there and holds all models of the reactor during one project build. In case Maven is being reused, the reactormodelpool is always refreshed. Having such pool is required for building the graph. This pool holds exactly the information required to get versions by path or groupId+artifactId.